streams/src/main/java/org/apache/kafka/streams/kstream/internals/InternalStreamsBuilder.java

```diff
@@ -194,4 +194,23 @@ public synchronized void addGlobalStore(final StoreBuilder<KeyValueStore> storeB
                                                processorName,
                                                stateUpdateSupplier);
     }
+    public synchronized void addGlobalStore(final StoreBuilder<KeyValueStore> storeBuilder,
+                                            final String topic,
+                                            final ConsumedInternal consumed,
+                                            final ProcessorSupplier stateUpdateSupplier) {
+        // explicitly disable logging for global stores
```

Comment:
we should not copy the code from the old `addGlobalStore()` but rather call the old `addGlobalStore()` passing the generated names.